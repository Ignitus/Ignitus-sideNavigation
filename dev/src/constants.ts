export const navItems = [
  {
    title: 'List Item Heading 1',
    children: [
      {
        title: 'Sub List Item 1',
        route: '/SubListItem1',
      },
      {
        title: 'Sub List Item 2',
        route: '/SubListItem2',
      },
      {
        title: 'Sub List Item 3',
        children: [
          {
            title: 'Sub Sub List Item 1',
            route: '/SubSubListItem1',
          },
          {
            title: 'Sub Sub List Item 2',
            route: '/SubSubListItem2',
          },
        ],
      },
    ],
  },
  {
    title: 'List Item Heading 2',
    children: [
      {
        title: 'Sub List Item 1',
        route: '/SubListItem1',
      },
      {
        title: 'Sub List Item 2',
        route: '/SubListItem2',
      },
      {
        title: 'Sub List Item 3',
        children: [
          {
            title: 'Sub Sub List Item 1',
            route: '/SubSubListItem1',
          },
          {
            title: 'Sub Sub List Item 2',
            route: '/SubSubListItem2',
          },
          {
            title: 'Sub Sub List Item 3',
            children: [
              {
                title: 'Sub Sub Sub List Item 1',
                route: '/SubSubSubListItem1',
              },
              {
                title: 'Sub Sub Sub List Item 2',
                route: '/SubSubSubListItem2',
              },
            ],
          },
        ],
      },
    ],
  },
];
