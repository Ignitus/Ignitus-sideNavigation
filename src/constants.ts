import { DarkBlue, Black, White, WhiteLilac, Red } from './helpers/colors';
import { ThemeWrapperProps } from './components/sideNavigation/component';

export const defaultTheme: ThemeWrapperProps = {
  navBackground: WhiteLilac,
  theme: {
    homeLinkColor: Black,
    listItemHeadingColor: DarkBlue,
    listItemHeadingArrowColor: DarkBlue,
    subListItemTextColor: Black,
    subListItemHeadingColor: Black,
    subListItemHeadingArrowColor: Red,
    hover: {
      subListItemBackgroundOnHover: DarkBlue,
      subListItemColorOnHover: White,
    },
  },
};
