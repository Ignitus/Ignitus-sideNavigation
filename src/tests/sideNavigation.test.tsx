import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import emotionSerializer, { matchers } from 'jest-emotion';
import renderer from 'react-test-renderer';

import { SideNavigation, ThemeWrapperProps } from '../index';
import { navItems } from './constants';

expect.addSnapshotSerializer(emotionSerializer);
expect.extend(matchers);

const mockColors = ['#000', '#fff', '#ff0000', '#000066', '#2D2D2D'];

const theme: ThemeWrapperProps = {
  navBackground: mockColors[0],
  theme: {
    homeLinkColor: mockColors[0],
    listItemHeadingColor: mockColors[3],
    listItemHeadingArrowColor: mockColors[3],
    subListItemTextColor: mockColors[0],
    subListItemHeadingColor: mockColors[0],
    subListItemHeadingArrowColor: mockColors[2],
    hover: {
      subListItemBackgroundOnHover: mockColors[3],
      subListItemColorOnHover: mockColors[1],
    },
  },
};

describe('<SideNavigation />', () => {
  it('should render', () => {
    const component = renderer.create(
      <Router>
        <SideNavigation navItems={navItems} />
      </Router>,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with theme', () => {
    const component = renderer.create(
      <Router>
        <SideNavigation navItems={navItems} theme={theme.theme} />
      </Router>,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
