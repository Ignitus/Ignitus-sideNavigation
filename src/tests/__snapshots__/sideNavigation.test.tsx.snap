// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideNavigation /> should render 1`] = `
.emotion-16 {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  height: 100vh;
  overflow-y: auto;
  overflow-x: auto;
  background-color: #fafbfd;
}

@media (max-width:768px) {
  .emotion-16 nav {
    position: fixed;
    overflow-y: auto;
    z-index: 1;
    background-color: #fafbfd;
  }
}

@media (max-width:420px) {
  .emotion-16 {
    height: auto;
    overflow: auto;
  }
}

@media (min-width:1024px) {
  .emotion-16 {
    -webkit-flex: 1.5;
    -ms-flex: 1.5;
    flex: 1.5;
  }
}

@media (min-width:1466px) {
  .emotion-16 {
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
  }
}

.emotion-15 {
  width: 100%;
  padding: 0 1rem;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-1 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #000;
}

.emotion-1 a:hover {
  color: #000066;
}

.emotion-0 {
  height: 2rem;
  margin-left: auto;
  display: none;
  margin-right: 2rem;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  -webkit-transition: -webkit-transform 200ms ease-in-out;
  -webkit-transition: transform 200ms ease-in-out;
  transition: transform 200ms ease-in-out;
}

@media (max-width:768px) {
  .emotion-0 {
    display: block;
  }
}

.emotion-14 {
  display: block;
}

@media (max-width:768px) {
  .emotion-14 {
    height: 100vh;
  }
}

@media (min-width:768px) {
  .emotion-14 {
    display: block;
  }
}

.emotion-2 {
  opacity: 0.5;
  margin: 1rem 0;
}

.emotion-3 {
  padding: 0.5rem;
  margin-left: 0rem;
  color: #000;
  cursor: pointer;
}

.emotion-3:hover {
  background-color: #000066;
  color: #ffffff;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 1rem;
  margin-bottom: 0.5rem;
  margin-left: 0rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.emotion-4 {
  font-size: 22px;
  color: #000066;
  font-weight: 500;
  padding: 0;
  margin: 0;
}

.emotion-5 {
  height: 2rem;
  fill: #000066;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  -webkit-transition: -webkit-transform 200ms ease-in-out;
  -webkit-transition: transform 200ms ease-in-out;
  transition: transform 200ms ease-in-out;
}

.emotion-13 {
  list-style: none;
  display: block;
  padding: 0;
  margin: 0;
}

.emotion-13 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #000;
}

.emotion-7 {
  padding: 0.5rem;
  margin-left: 0.3333333333333333rem;
  color: #000;
  cursor: pointer;
}

.emotion-7:hover {
  background-color: #000066;
  color: #ffffff;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 0;
  margin-bottom: 0;
  margin-left: 0.3333333333333333rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.emotion-8 {
  font-size: 16px;
  color: #000;
  font-weight: 400;
  padding: 0.5rem;
  margin: 0;
}

.emotion-9 {
  height: 1.5rem;
  fill: #ff0000;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: -webkit-transform 200ms ease-in-out;
  -webkit-transition: transform 200ms ease-in-out;
  transition: transform 200ms ease-in-out;
}

.emotion-12 {
  list-style: none;
  display: none;
  padding: 0;
  margin: 0;
}

.emotion-12 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #000;
}

.emotion-11 {
  padding: 0.5rem;
  margin-left: 0.6666666666666666rem;
  color: #000;
  cursor: pointer;
}

.emotion-11:hover {
  background-color: #000066;
  color: #ffffff;
}

<div
  className="emotion-16"
>
  <nav
    className="emotion-15"
  >
    <h1
      className="emotion-1"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home ðŸ‘‹
      </a>
      <svg
        aria-hidden="true"
        aria-label="Arrow"
        className="emotion-0"
        data-icon="Arrow"
        onClick={[Function]}
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Arrow
        </title>
        <path
          d="M9.29 15.88L13.17 12 9.29 8.12c-.39-.39-.39-1.02 0-1.41.39-.39 1.02-.39 1.41 0l4.59 4.59c.39.39.39 1.02 0 1.41L10.7 17.3c-.39.39-1.02.39-1.41 0-.38-.39-.39-1.03 0-1.42z"
        />
      </svg>
    </h1>
    <div
      className="emotion-14"
    >
      <hr
        className="emotion-2"
      />
      <a
        href="/"
        onClick={[Function]}
      >
        <li
          className="emotion-3"
        >
          H1
        </li>
      </a>
      <div
        className="emotion-6"
        onClick={[Function]}
      >
         
        <h5
          className="emotion-4"
        >
          H2
        </h5>
         
        <svg
          aria-hidden="true"
          aria-label="Arrow"
          className="emotion-5"
          data-icon="Arrow"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Arrow
          </title>
          <path
            d="M9.29 15.88L13.17 12 9.29 8.12c-.39-.39-.39-1.02 0-1.41.39-.39 1.02-.39 1.41 0l4.59 4.59c.39.39.39 1.02 0 1.41L10.7 17.3c-.39.39-1.02.39-1.41 0-.38-.39-.39-1.03 0-1.42z"
          />
        </svg>
      </div>
      <ul
        className="emotion-13"
      >
        <a
          href="/h2"
          onClick={[Function]}
        >
          <li
            className="emotion-7"
          >
            H2.1
          </li>
        </a>
        <div
          className="emotion-10"
          onClick={[Function]}
        >
           
          <h5
            className="emotion-8"
          >
            H2.2
          </h5>
           
          <svg
            aria-hidden="true"
            aria-label="Arrow"
            className="emotion-9"
            data-icon="Arrow"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Arrow
            </title>
            <path
              d="M9.29 15.88L13.17 12 9.29 8.12c-.39-.39-.39-1.02 0-1.41.39-.39 1.02-.39 1.41 0l4.59 4.59c.39.39.39 1.02 0 1.41L10.7 17.3c-.39.39-1.02.39-1.41 0-.38-.39-.39-1.03 0-1.42z"
            />
          </svg>
        </div>
        <ul
          className="emotion-12"
        >
          <a
            href="/h21"
            onClick={[Function]}
          >
            <li
              className="emotion-11"
            >
              H2.2.1
            </li>
          </a>
        </ul>
      </ul>
    </div>
  </nav>
</div>
`;

exports[`<SideNavigation /> should render with theme 1`] = `
.emotion-16 {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  height: 100vh;
  overflow-y: auto;
  overflow-x: auto;
  background-color: #fafbfd;
}

@media (max-width:768px) {
  .emotion-16 nav {
    position: fixed;
    overflow-y: auto;
    z-index: 1;
    background-color: #fafbfd;
  }
}

@media (max-width:420px) {
  .emotion-16 {
    height: auto;
    overflow: auto;
  }
}

@media (min-width:1024px) {
  .emotion-16 {
    -webkit-flex: 1.5;
    -ms-flex: 1.5;
    flex: 1.5;
  }
}

@media (min-width:1466px) {
  .emotion-16 {
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
  }
}

.emotion-15 {
  width: 100%;
  padding: 0 1rem;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-1 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #000;
}

.emotion-1 a:hover {
  color: #000066;
}

.emotion-0 {
  height: 2rem;
  margin-left: auto;
  display: none;
  margin-right: 2rem;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  -webkit-transition: -webkit-transform 200ms ease-in-out;
  -webkit-transition: transform 200ms ease-in-out;
  transition: transform 200ms ease-in-out;
}

@media (max-width:768px) {
  .emotion-0 {
    display: block;
  }
}

.emotion-14 {
  display: block;
}

@media (max-width:768px) {
  .emotion-14 {
    height: 100vh;
  }
}

@media (min-width:768px) {
  .emotion-14 {
    display: block;
  }
}

.emotion-2 {
  opacity: 0.5;
  margin: 1rem 0;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 1rem;
  margin-bottom: 0.5rem;
  margin-left: 0rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.emotion-4 {
  font-size: 22px;
  color: #000066;
  font-weight: 500;
  padding: 0;
  margin: 0;
}

.emotion-5 {
  height: 2rem;
  fill: #000066;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  -webkit-transition: -webkit-transform 200ms ease-in-out;
  -webkit-transition: transform 200ms ease-in-out;
  transition: transform 200ms ease-in-out;
}

.emotion-13 {
  list-style: none;
  display: block;
  padding: 0;
  margin: 0;
}

.emotion-13 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #000;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 0;
  margin-bottom: 0;
  margin-left: 0.3333333333333333rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.emotion-8 {
  font-size: 16px;
  color: #000;
  font-weight: 400;
  padding: 0.5rem;
  margin: 0;
}

.emotion-9 {
  height: 1.5rem;
  fill: #ff0000;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: -webkit-transform 200ms ease-in-out;
  -webkit-transition: transform 200ms ease-in-out;
  transition: transform 200ms ease-in-out;
}

.emotion-12 {
  list-style: none;
  display: none;
  padding: 0;
  margin: 0;
}

.emotion-12 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #000;
}

.emotion-3 {
  padding: 0.5rem;
  margin-left: 0rem;
  color: #000;
  cursor: pointer;
}

.emotion-3:hover {
  background-color: #000066;
  color: #fff;
}

.emotion-7 {
  padding: 0.5rem;
  margin-left: 0.3333333333333333rem;
  color: #000;
  cursor: pointer;
}

.emotion-7:hover {
  background-color: #000066;
  color: #fff;
}

.emotion-11 {
  padding: 0.5rem;
  margin-left: 0.6666666666666666rem;
  color: #000;
  cursor: pointer;
}

.emotion-11:hover {
  background-color: #000066;
  color: #fff;
}

<div
  className="emotion-16"
>
  <nav
    className="emotion-15"
  >
    <h1
      className="emotion-1"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home ðŸ‘‹
      </a>
      <svg
        aria-hidden="true"
        aria-label="Arrow"
        className="emotion-0"
        data-icon="Arrow"
        onClick={[Function]}
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Arrow
        </title>
        <path
          d="M9.29 15.88L13.17 12 9.29 8.12c-.39-.39-.39-1.02 0-1.41.39-.39 1.02-.39 1.41 0l4.59 4.59c.39.39.39 1.02 0 1.41L10.7 17.3c-.39.39-1.02.39-1.41 0-.38-.39-.39-1.03 0-1.42z"
        />
      </svg>
    </h1>
    <div
      className="emotion-14"
    >
      <hr
        className="emotion-2"
      />
      <a
        href="/"
        onClick={[Function]}
      >
        <li
          className="emotion-3"
        >
          H1
        </li>
      </a>
      <div
        className="emotion-6"
        onClick={[Function]}
      >
         
        <h5
          className="emotion-4"
        >
          H2
        </h5>
         
        <svg
          aria-hidden="true"
          aria-label="Arrow"
          className="emotion-5"
          data-icon="Arrow"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Arrow
          </title>
          <path
            d="M9.29 15.88L13.17 12 9.29 8.12c-.39-.39-.39-1.02 0-1.41.39-.39 1.02-.39 1.41 0l4.59 4.59c.39.39.39 1.02 0 1.41L10.7 17.3c-.39.39-1.02.39-1.41 0-.38-.39-.39-1.03 0-1.42z"
          />
        </svg>
      </div>
      <ul
        className="emotion-13"
      >
        <a
          href="/h2"
          onClick={[Function]}
        >
          <li
            className="emotion-7"
          >
            H2.1
          </li>
        </a>
        <div
          className="emotion-10"
          onClick={[Function]}
        >
           
          <h5
            className="emotion-8"
          >
            H2.2
          </h5>
           
          <svg
            aria-hidden="true"
            aria-label="Arrow"
            className="emotion-9"
            data-icon="Arrow"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Arrow
            </title>
            <path
              d="M9.29 15.88L13.17 12 9.29 8.12c-.39-.39-.39-1.02 0-1.41.39-.39 1.02-.39 1.41 0l4.59 4.59c.39.39.39 1.02 0 1.41L10.7 17.3c-.39.39-1.02.39-1.41 0-.38-.39-.39-1.03 0-1.42z"
            />
          </svg>
        </div>
        <ul
          className="emotion-12"
        >
          <a
            href="/h21"
            onClick={[Function]}
          >
            <li
              className="emotion-11"
            >
              H2.2.1
            </li>
          </a>
        </ul>
      </ul>
    </div>
  </nav>
</div>
`;
